xarray/tests/test_backends.py

```diff
@@ -2213,6 +2212,77 @@ class PyNioTestAutocloseTrue(PyNioTest):
     autoclose = True
 
 
+@requires_netCDF4
+@requires_pnc
+class TestPseudoNetCDF(CFEncodedDataTest, NetCDF3Only, TestCase):
+    def test_write_store(self):
+        # pseudonetcdf is read-only for now
+        pass
+
+    @contextlib.contextmanager
+    def open(self, path, **kwargs):
+        with open_dataset(path, engine='pseudonetcdf',
+                          autoclose=self.autoclose,
+                          **kwargs) as ds:
+            yield ds
+
+    def save(self, dataset, path, **kwargs):
+        dataset.to_netcdf(path, engine='netcdf4', **kwargs)
+
+    def test_weakrefs(self):
+        example = Dataset({'foo': ('x', np.arange(5.0))})
+        expected = example.rename({'foo': 'bar', 'x': 'y'})
+
+        with create_tmp_file() as tmp_file:
+            example.to_netcdf(tmp_file, engine='netcdf4')
+            on_disk = open_dataset(tmp_file, engine='pseudonetcdf')
+            actual = on_disk.rename({'foo': 'bar', 'x': 'y'})
+            del on_disk  # trigger garbage collection
+            assert_identical(actual, expected)
+
+    @pytest.mark.skip(reason='cannot pickle file objects')
+    def test_pickle(self):
+        pass
+
+    @pytest.mark.skip(reason='cannot pickle file objects')
+    def test_pickle_dataarray(self):
+        pass
+
+    @pytest.mark.skip(reason='does not support boolean dtype')
+    def test_roundtrip_boolean_dtype(self):
+        # pnc does not support boolean dtype
+        pass
+
+    @pytest.mark.skip(reason='does not support auto masking and scaling')
+    def test_roundtrip_mask_and_scale(self):
+        # pnc does not support auto masking and scaling
```

Comment:
Again, this is done in the xarray layer, by parsing the `_FillValue`, `scale_factor` and `add_offset` attributes.

Does pnc handle masking of data in some way? Do any of the file formats it works with have some way of marking data as invalid?

Reply:
The format that is being used by default is NetCDF4. Oddly, it seems to fail very early on in the test. It is not obvious to me where pnc is even used yet.

```
        decoded = create_masked_and_scaled_data()
        encoded = create_encoded_masked_and_scaled_data()
        with self.roundtrip(decoded) as actual:
            assert_allclose(decoded, actual, decode_bytes=False)
```

Reply:
Okay, I think I get it. PNC allows use of netCDF4, but netCDF4 has to disable auto mask and scale to work with xarray.

Most formats supported by pnc (uamiv, ffi1001, bpch, arl packedbit) do not use CF masking and scaling. I can add a test-case file for one of these formats, but I'd still just end up disabling the test.

Please advise.